google/resource_task_queue.go

```diff
@@ -0,0 +1,241 @@
+							Optional: true,
+							Computed: true,
+						},
+						"max_doublings": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"max_backoff": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateRegexp(`^[0-9]*s$`),
+						},
+						"min_backoff": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateRegexp(`^[0-9]*\.?[0-9]+s$`),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceTaskQueueRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	location := d.Get("location")
+	name := d.Get("name")
+
+	url := fmt.Sprintf("projects/%s/locations/%s/queues/%s", project, location, name)
+
+	resp, err := config.clientCloudTasks.Projects.Locations.Queues.Get(url).Do()
+	if err != nil {
+		return fmt.Errorf("Error reading task queue: %s %s", location, err)
+	}
+
+	d.Set("retry", flattenRetryConfig(resp.RetryConfig))
+	d.Set("rate_limits", flattenRateLimits(resp.RateLimits))
+
+	return nil
+}
+
+func resourceTaskQueueCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	location := d.Get("location")
+	name := d.Get("name")
+
+	url := fmt.Sprintf("projects/%s/locations/%s/queues/%s", project, location, name)
+	id := fmt.Sprintf("%s/%s/%s", project, location, name)
+
+	rb := &cloudtasks.Queue{
+		Name:        url,
+		RateLimits:  expandRateLimits(d.Get("rate_limits").([]interface{})),
+		RetryConfig: expandRetryConfig(d.Get("retry").([]interface{})),
+	}
+
+	log.Printf("[DEBUG] Updating Task Queue: %#v", name)
+
+	resp, err := config.clientCloudTasks.Projects.Locations.Queues.Patch(url, rb).Do()
+	if err != nil {
+		return fmt.Errorf("Error updating task queue %s: %s", name, err)
+	}
+
+	d.SetId(id)
+
+	log.Printf("[DEBUG] Finished updating new Task Queue: %#v\n%#v\n", name, resp)
+
+	return resourceTaskQueueRead(d, meta)
+}
+
+func resourceTaskQueueDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	location := d.Get("location")
+	name := d.Get("name")
+
+	url := fmt.Sprintf("projects/%s/locations/%s/queues/%s", project, location, name)
+
+	log.Printf("[DEBUG] Deleting Task Queue: %#v", name)
+
+	resp, err := config.clientCloudTasks.Projects.Locations.Queues.Delete(url).Do()
+	if err != nil {
+		return fmt.Errorf("Error deleting task queue: %s", err)
+	}
+
+	log.Printf("[DEBUG] Finished deleting Task Queue: %#v\n%#v\n", name, resp)
+
+	return nil
+}
+
+func resourceTaskQueueImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
```

Comment:
Consider doing what resource_kms_key_ring.go does for its import, using parseImportId() and replaceVars().